src/rgw/rgw_crypt.h

```diff
@@ -134,7 +137,7 @@ class RGWPutObj_BlockEncrypt : public rgw::putobj::Pipe
                          rgw::sal::DataProcessor *next,
                          std::unique_ptr<BlockCrypt> crypt);
 
-  int process(bufferlist&& data, uint64_t logical_offset) override;
+  int process(bufferlist&& data, uint64_t logical_offset, optional_yield y) override;
```

Comment:
i think we can avoid changing the base `rgw::sal::DataProcessor` interface by passing this `optional_yield` into `RGWPutObj_BlockEncrypt`'s constructor instead

Reply:
> i think we can avoid changing the base `rgw::sal::DataProcessor` interface by passing this `optional_yield` into `RGWPutObj_BlockEncrypt`'s constructor instead

I am done based this. And I have removed all `optional_yield` in process() function both src/rgw and test/rgw.